api/management/commands/create_groups.py

```diff
@@ -17,15 +22,21 @@ def add_arguments(self, parser: CommandParser) -> None:
         parser.add_argument("--teacher_ids", nargs="*", type=int)
 
     def handle(self, *_: str, **options: Any) -> None:
-        teachers: Iterable[Teacher] = []
+        teachers: Iterable[Teacher]
         if options["teacher_ids"]:
-            teachers = Teacher.objects.filter(pk__in=options["teacher_ids"])
-            for teacher in teachers:
```

Comment:
maybe factor out this loop into a small function to remove the loop from within the if? Just for the sake of readability and maybe also to have operations of the same level of granularity in a function (the else-clause contains a call to a method)

Reply:
(nit, although I do like same level of granularity 😆)

Reply:
I agree here, changing this. It will also remove that weird type annotation you mentioned on top (yes, it made the difference for one of the linters, apparently)